src/main/java/io/kestra/plugin/aws/kinesis/Consume.java

```diff
@@ -0,0 +1,207 @@
+
+        File outFile = runContext.workingDir().createTempFile(".ion").toFile();
+        BufferedOutputStream output = new BufferedOutputStream(new FileOutputStream(outFile));
+
+        int consumed = 0;
+        Map<String, String> lastSequence = new HashMap<>();
+        Instant deadline = Instant.now().plus(runContext.render(maxDuration).as(Duration.class).orElseThrow());
+
+        for (Shard shard : shards) {
+            String iterator = buildShardIterator(runContext, client, rStream, shard);
+
+            while (iterator != null) {
+                GetRecordsResponse response = client.getRecords(
+                    GetRecordsRequest.builder()
+                        .shardIterator(iterator)
+                        .limit(1000)
+                        .build()
+                );
+
+                for (Record record : response.records()) {
+                    ConsumedRecord consumedRecord = ConsumedRecord.builder()
+                        .data(new String(record.data().asByteArray()))
+                        .partitionKey(record.partitionKey())
+                        .sequenceNumber(record.sequenceNumber())
+                        .approximateArrivalTimestamp(record.approximateArrivalTimestamp())
+                        .shardId(shard.shardId())
+                        .build();
+
+                    FileSerde.write(output, consumedRecord);
+
+                    consumed++;
+                    lastSequence.put(shard.shardId(), record.sequenceNumber());
+
+                    if (consumed >= runContext.render(maxRecords).as(Integer.class).orElse(Integer.MAX_VALUE)) {
+                        break;
+                    }
+                }
+
+                iterator = response.nextShardIterator();
+
+                if (Instant.now().isAfter(deadline) ||
+                    consumed >= runContext.render(maxRecords).as(Integer.class).orElse(Integer.MAX_VALUE)) {
+                    break;
+                }
+
+                Thread.sleep(runContext.render(pollDuration).as(Duration.class).orElse(Duration.ofSeconds(1)).toMillis());
+            }
+        }
+
+        output.flush();
+        URI uri = runContext.storage().putFile(outFile);
+
+        // metrics
+        runContext.metric(Timer.of("duration", Duration.ofNanos(System.nanoTime() - startedAt)));
+        runContext.metric(Counter.of("records", consumed));
+
+        return Output.builder()
+            .uri(uri)
+            .count(consumed)
+            .lastSequencePerShard(lastSequence)
+            .build();
+    }
+
+
+    private String buildShardIterator(RunContext runContext, KinesisClient client, String stream, Shard shard) throws IllegalVariableEvaluationException {
+        var builder = GetShardIteratorRequest.builder()
+            .streamName(stream)
+            .shardId(shard.shardId())
+            .shardIteratorType(ShardIteratorType.fromValue(
+                runContext.render(iteratorType).as(String.class).orElseThrow()
```

Comment:
And even better with an enum after your refactor:

```suggestion
                runContext.render(iteratorType).as(String.class).orElse("LATEST")
```